### DISCUSSION TITLE
Feature Request: Add support for reading/writing files in HDFS

### DISCUSSION THREAD

**Discussion by rotem-ad** [duckdb/duckdb#8131] [Ideas]
Similarly to the S3 API support (provided by the [httpfs extension](https://duckdb.org/docs/extensions/httpfs)), it'd be super useful to have the option to access Parquet/CSV files stored in HDFS.
For example:
```
SELECT * from parquet_scan('hdfs://hdfs-namenode/my_dataset/*.parquet');
```
I strongly believe adding this feature will significantly increase the popularity and usage of DuckDB in the community.

**Comment by Tishj** [comment]
I can't give you any answers on whether or not we'll add support for this,

But in the meantime you should be able to use duckdb together with HDFS through the `fsspec` library in Python.

See the docs for a good introduction to using `fsspec` together with DuckDB.
https://duckdb.org/docs/guides/python/filesystems

**Comment by georgelza** [comment]
i will second this request, being able to use duckdb to get to both iceberg on s3 or paimon on hdfs would be amazing, one interface layer... with both table formats... storing as parquet file format.

**Comment by rotem-ad** [comment]
@georgelza Totally agreed. It'd be amazing to have an official HDFS extension for DuckDB!
Btw, I happened to find this unofficial HDFS extension on GitHub: https://github.com/vincent-chang/duckdb-hdfs/tree/main. Didn't have the chance to test it yet

**Comment by georgelza** [comment]
need to see... wondering... as you can use duckdb to query parquet tables on s3 written as iceberg tables... would it be able to query parquet tables on s3 written as paimon as tables...
it's still duckdb and s3...

**Comment by 54kggo** [comment]
use fsspec is a good way to read and write hdfs files with duckdb. [Using fsspec Filesystems](https://duckdb.org/docs/guides/python/filesystems)
I use webhdfs protocol to solve this problem
`
import duckdb

from fsspec.implementations.webhdfs import WebHDFS

duckdb.register_filesystem(WebHDFS(host="xx.xx.xx.xx", port=1234))

duckdb.sql("COPY (SELECT * FROM read_csv('webhdfs:///xxx/xxx/xxx.csv')) TO 'webhdfs:///xxx/xxx/xxx.parquet' (FORMAT PARQUET)")
`

**Comment by georgelza** [comment]
Been thinking...

S3 and HDFS read capability.
Integration with Hive Metastore and ..., releasing the analyst from needing
to tell it which files to stick together...

The power sits with making it as simple for a data analyst, and trust me
they don't care for manifest files, snapshots various versions of the
parquet files...

G

**Comment by RoeyoOgen** [comment]
any update on this?
